etcd/server/service.sls

```diff
@@ -1,13 +1,48 @@
 {%- from "etcd/map.jinja" import server with context %}
 {%- if server.enabled %}
 
+# get binaries and install packages
 etcd_packages:
   pkg.installed:
-  - names: {{ server.pkgs }}
-{%- if server.get('engine', 'systemd') %}
-  - require:
-    - file: /etc/default/etcd
-{%- endif %}
+    - names: {{ server.pkgs }}
+
+/tmp/etcd/:
```

Comment:
If you really want to do something so dirty like this (eg. aren't binaries downloadable from somewhere else?), introduce shell script that will be called from salt formula and will take care of everything (starting container, taking binaries from it), etc. and will therefore solve idempotency.